modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -4584,37 +4683,54 @@ public CodegenOperation fromOperation(String path,
         CodegenParameter bodyParam = null;
         RequestBody requestBody = ModelUtils.getReferencedRequestBody(this.openAPI, operation.getRequestBody());
         if (requestBody != null) {
-            String contentType = getContentType(requestBody);
-            if (contentType != null) {
-                contentType = contentType.toLowerCase(Locale.ROOT);
-            }
-            if (contentType != null &&
-                    ((!(this instanceof RustAxumServerCodegen) && contentType.startsWith("application/x-www-form-urlencoded")) ||
-                            contentType.startsWith("multipart"))) {
-                // process form parameters
-                formParams = fromRequestBodyToFormParameters(requestBody, imports);
-                op.isMultipart = contentType.startsWith("multipart");
-                for (CodegenParameter cp : formParams) {
-                    setParameterEncodingValues(cp, requestBody.getContent().get(contentType));
-                    postProcessParameter(cp);
-                }
-                // add form parameters to the beginning of all parameter list
-                if (prependFormOrBodyParameters) {
+            LOGGER.debug("Op: " + op.operationId + " has request body");
+
+            requestBody = ModelUtils.getReferencedRequestBody(this.openAPI, requestBody);
+
+            Map<String, Schema> bodySchemas = ModelUtils.getSchemasFromRequestBody(requestBody);
+
+            boolean simpleForm = false;
+
+            if (bodySchemas != null && (bodySchemas.size() == 1 || !supportsMultipleRequestTypes)) {
+                LOGGER.debug("Op: " + op.operationId + " has request body with single schema");
+
+                Map.Entry<String, Schema> entry = bodySchemas.entrySet().iterator().next();
+
+                if (!(this instanceof RustAxumServerCodegen) && isForm(entry.getKey())) {
```

Comment:
in default codegen (this file). some conditions use `this instanceof RustServerCodegen` and we would like to remove these if possible.

not something we need to do in this PR as this PR is huge already.

more of an FYI

Reply:
Yeah - this was existing. I think we want something like `useExplodedFormHandling` or similar.

Reply:
I've raised https://github.com/OpenAPITools/openapi-generator/issues/19623 to cover this.